src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -73,7 +79,7 @@ public function populate(array $values): void
         foreach ($values as $name => $value) {
             $notHttpName = 0 !== strpos($name, 'HTTP_');
             // don't check existence with getenv() because of thread safety issues
-            if (!isset($loadedVars[$name]) && (isset($_ENV[$name]) || (isset($_SERVER[$name]) && $notHttpName))) {
+            if (!isset($loadedVars[$name]) && ((!$this->enableOverride && isset($_ENV[$name])) || (isset($_SERVER[$name]) && $notHttpName))) {
```

Comment:
Shouldn't this be used for all `isset` checks?

```php
if (!isset($loadedVars[$name]) && (!$this->enableOverride && (isset($_ENV[$name]) || (isset($_SERVER[$name]) && $notHttpName)))) {
```

Additionally, I prefer an explicit `=== false` check for readability, but I'm not sure what the convention is.

Reply:
For security reasons I think it's better to not allow overriding `$_SERVER['HTTP_*']` values, a previous discussion took place before the creation of this component. Maybe a core team member can confirm this later.

About CS style, there are plenty of examples in the codebase (and in this file too) where shortcut is used. Personaly I think it's very strange to explicitly test a false value against a boolean return type :)

Reply:
The `!$notHttpName`  already fulfills the stop condition to not overwrite the http server vars. So I think it's safe to include your check on the `$_SERVER` ?

Reply:
Updated